### PR TITLE
Unique email

### PR DESCRIPTION
**Pull request by MarinaIvanova-1**
We couldn't make a unit test for the email uniqueness, but the integration test works as expected. Created a separate ticket for the unit test on Trello - it might need some research. Integration sign up test has been implemented with generating a random email every time so it doesn't repeat as we couldn't figure out how to drop the collection in the database before the test. For now all integration tests are passing.

### CODE REVIEW

**Comment by MarinaIvanova-1** [comment]
[(M) Error handling for duplicate email](https://trello.com/c/iBvMmjWu/23-m-error-handling-for-duplicate-email)

**Comment by MarinaIvanova-1** [comment]
[(M) Email needs to be Unique](https://trello.com/c/jLWCkXzo/18-m-email-needs-to-be-unique)

**Comment by MarinaIvanova-1** [comment]
[(S) Tests for navbar - fix broken new post test](https://trello.com/c/NZlp2n0m/30-s-tests-for-navbar-fix-broken-new-post-test)

File: cypress/integration/user_can_sign_up_spec.js [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+        }
+     return result;
+    }
+
+    const randomEmail = `${fakeEmail(10)}@example.com`
+
     cy.visit("/users/new");
     cy.get("#name").type("someone");
-    cy.get("#email").type("someone@example.com");
+    cy.get("#email").type(randomEmail);
```
Comment (44jovi) [inline cypress/integration/user_can_sign_up_spec.js:19]: Really like this temp workaround - great idea

File: cypress/integration/user_cant_sign_up_with_the_same_email_spec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+
+    cy.visit("/users/new");
+    cy.get("#name").type("someone");
+    cy.get("#email").type("someone@example.com");
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+
+    cy.url().should("include", "/users");
```
Comment (44jovi) [inline cypress/integration/user_cant_sign_up_with_the_same_email_spec.js:17]: For me this line fails in the Cypress tests (this is the only spec file that fails out of all 6).

Error message from Cypress results:

_AssertonError: Timed out retrying after 4000ms: expected 'http://localhost:3030/sessions/new' to include '/users'_

Also tried signing up with existing email in browser using test server and it does just redirect me back to _/sessions/new_ and didn't spot anything odd in dev console.

We can check if it's same for anyone else.

File: controllers/users.js [hunk 3]
```diff
@@ -9,7 +9,12 @@ const UsersController = {
     const user = new User(req.body);
     user.save((err) => {
       if (err) {
-        throw err;
+        if (err.name === "MongoError" && err.code === 11000) {
+          // Duplicate email
+          return res
+            .status(422)
+            .render("error", { message: "This email already exists" });
+        }
```
Comment (44jovi) [inline controllers/users.js:17]: May need to review this - see comment on this PR for:
_cypress/integration/user_cant_sign_up_with_the_same_email_spec.js_

File: cypress/integration/user_cant_sign_up_with_the_same_email_spec.js (hunk 2 above)
Comment (44jovi) [reply cypress/integration/user_cant_sign_up_with_the_same_email_spec.js:17]: RESOLVED

**Pull request merged**